sw/device/lib/dif/dif_otp_ctrl.c

```diff
@@ -552,84 +405,84 @@ static const partition_info_t kPartitions[] = {
         },
 };
 
-dif_otp_ctrl_dai_result_t dif_otp_ctrl_dai_read_start(
-    const dif_otp_ctrl_t *otp, dif_otp_ctrl_partition_t partition,
-    uint32_t address) {
+dif_result_t dif_otp_ctrl_dai_read_start(const dif_otp_ctrl_t *otp,
+                                         dif_otp_ctrl_partition_t partition,
+                                         uint32_t address) {
   if (otp == NULL || partition >= ARRAYSIZE(kPartitions)) {
-    return kDifOtpCtrlDaiBadArg;
+    return kDifBadArg;
   }
 
   if ((address & kPartitions[partition].align_mask) != 0) {
-    return kDifOtpCtrlDaiUnaligned;
+    return kDifUnaligned;
```

Comment:
@sriyerg see here for context you asked about

Reply:
What if I want to intentionally make an unaligned / out of range access to OTP to verify that the HW is behaving correctly?

Reply:
I should also note that we haven't planned for making these types of accesses at the chip level - the assumption is the outcome is adequately verified at the OTP ctrl IP level. @cindychip, fya. 

Though, we may need to support this in future.

Reply:
I'm not up to date on the otp_ctrl test plan so I'm not sure if thats something we are trying to test for the nearest milestone. If that's something we **do** want to test, then obviously we **don't** want the DIF to stifle/error out on these accesses. If thats the case, then let's file a separate issue and PR to fix. Trying to keep this PR to just: a) setting up the IP's DIF structure to allow auto-generation of a subset of DIFs, and b) actually auto-generating a subset of (IRQ) DIFs.

Reply:
Agreed.